librosa/core/audio.py

```diff
@@ -914,22 +915,27 @@ def autocorrelate(
 
     fft = get_fftlib()
 
+    real = not np.iscomplexobj(y)
+
     # Pad out the signal to support full-length auto-correlation.
-    n_pad = 2 * y.shape[axis] - 1
+    if hasattr(scipy, "fft") and hasattr(scipy.fft, "next_fast_len"):
```

Comment:
This isn't quite right.  Older versions of scipy (specifically 1.2.0) had `scipy.fft` as a function, not a module, so the first hasattr check is not going to do anything here.

This is why I had proposed try-except instead of a hasattr check.

Reply:
Actually only `hasattr(scipy.fft, "next_fast_len")` is needed. I wrote the first condition to make my meaning clearer.

I personally avoid the use of try-except block as a normal control flow.

Reply:
Yes, in general hasattr is preferable to try-except for this kind of thing.  However, scipy does do some tricks with imports and submodule structure.  But if a single hasattr call actually does work in the older versions, that seems fine.

Reply:
We should remove the redundant hasattr check.  I think the code is clear enough without it.